bouncer/shared/contract_swap.ts

```diff
@@ -5,64 +5,135 @@ import {
+    ? u8aToHex(
+        vaultSwapCfParametersCodec.enc({
+          ccmAdditionalData: ccmAdditionalData ? hexToU8a(ccmAdditionalData) : undefined,
+          vaultSwapParameters:
+            fillOrKillParams || dcaParams || boostFeeBps
+              ? {
+                  refundParams: fillOrKillParams && {
+                    retryDurationBlocks: fillOrKillParams.retryDurationBlocks,
+                    refundAddress: {
+                      tag: shortChainFromAsset(sourceAsset),
+                      value: hexToU8a(fillOrKillParams.refundAddress),
+                    },
+                    minPriceX128: BigInt(fillOrKillParams.minPriceX128),
+                  },
+                  dcaParams,
+                  boostFee: boostFeeBps,
+                }
+              : undefined,
+        }),
+      )
+    : undefined;
+}
+
 export async function executeContractSwap(
-  srcAsset: Asset,
+  sourceAsset: Asset,
   destAsset: Asset,
   destAddress: string,
   wallet: HDNodeWallet,
   messageMetadata?: CcmDepositMetadata,
+  amount?: string,
+  boostFeeBps?: number,
+  fillOrKillParams?: FillOrKillParamsX128,
+  dcaParams?: DcaParams,
 ): ReturnType<typeof executeSwap> {
-  const srcChain = chainFromAsset(srcAsset);
+  const srcChain = chainFromAsset(sourceAsset);
   const destChain = chainFromAsset(destAsset);
+  const amountToSwap = amount ?? defaultAssetAmounts(sourceAsset);
 
   const networkOptions = {
     signer: wallet,
     network: 'localnet',
     vaultContractAddress: getContractAddress(srcChain, 'VAULT'),
-    srcTokenContractAddress: getContractAddress(srcChain, srcAsset),
+    srcTokenContractAddress: getContractAddress(srcChain, sourceAsset),
   } as const;
   const txOptions = {
     // This is run with fresh addresses to prevent nonce issues. Will be 1 for ERC20s.
     gasLimit: srcChain === Chains.Arbitrum ? 10000000n : 200000n,
   } as const;
 
+  const cfParameters = encodeCfParameters(
+    sourceAsset,
+    messageMetadata?.cfParameters,
+    boostFeeBps,
+    fillOrKillParams,
+    dcaParams,
+  );
+
   const receipt = await executeSwap(
     {
       destChain,
       destAsset: stateChainAssetFromAsset(destAsset),
       // It is important that this is large enough to result in
       // an amount larger than existential (e.g. on Polkadot):
-      amount: amountToFineAmount(defaultAssetAmounts(srcAsset), assetDecimals(srcAsset)),
+      amount: amountToFineAmount(amountToSwap, assetDecimals(sourceAsset)),
       destAddress,
-      srcAsset: stateChainAssetFromAsset(srcAsset),
+      srcAsset: stateChainAssetFromAsset(sourceAsset),
       srcChain,
+      // TODO: This will need some refactoring either putting the cfParameters outside the
+      // ccmParams and the user should encode it as done here or, probably better, we just
+      // add the SwapParameters support (Fok/Dca/Boost) as separate parameters, rename
+      // cfParameters to ccmAdditionalData and do the encoding within the SDK.
```

Comment:
I'll move the `encodeCfParameters` into the SDK and then pass the DCA/Boost/Fok parameters separately here, as for deposit channels, so we abstract that layer of complexity from the end user.